lib/segment/src/index/field_index/full_text_index/text_index.rs

```diff
@@ -213,21 +213,44 @@ impl FullTextIndex {
         bincode::deserialize(data).unwrap()
     }
 
-    pub(super) fn serialize_document_tokens(tokens: BTreeSet<String>) -> OperationResult<Vec<u8>> {
+    pub(super) fn serialize_token_set(tokens: BTreeSet<String>) -> OperationResult<Vec<u8>> {
         #[derive(Serialize)]
-        struct StoredDocument {
+        struct StoredTokens {
+            tokens: BTreeSet<String>,
+        }
+        let doc = StoredTokens { tokens };
```

Comment:
Could you check if we can simply use a `Vec` here. IIRC, the serialized format is compatible. If possible, we probably want an assertion here that asserts ordering just in case.